src/middleware/auth.js

```diff
@@ -0,0 +1,82 @@
+const jwt = require('jsonwebtoken');
+
+const { ACCESS_TOKEN_SECRET, REFRESH_TOKEN_SECRET } = require('../config/config');
+
+const createNewRefreshToken = userID => {
+  return jwt.sign({ userId: userID }, REFRESH_TOKEN_SECRET, {
+    expiresIn: '7d',
+  });
+};
+
+const createNewAccessToken = userID => {
+  return jwt.sign({ userId: userID }, ACCESS_TOKEN_SECRET, {
+    expiresIn: '1h',
+  });
+};
+
+const authenticateToken = (req, res, next) => {
+  try {
+    const accessToken = req.cookies['access-token'];
+    if (!accessToken) {
+      // eslint-disable-next-line no-console
+      console.log('Missing access token');
+    }
+    const accessData = jwt.verify(accessToken, ACCESS_TOKEN_SECRET);
+    req.userId = accessData.userId;
+  } catch (error) {
+    if (error.name === 'TokenExpiredError') {
+      //Generate a refresh token
+      try {
+        const refreshToken = req.cookies['refresh-token'];
+        if (!refreshToken) {
+          console.log('Missing refresh token');
+          return res.status(404).json({ error: 'Missing refresh token' });
```

Comment:
srry about confusion, let's remove any `404` errors because it'll introduce bugs with the `/register` and `/login` endpoints. let's just always call `next()` at the end